app/js/components/orderEntry/addForm/DraftDataTable.jsx

```diff
@@ -2,70 +2,61 @@ import React from 'react';
+  orders,
+  handleEditDrugOrder,
+  handleDiscardOneOrder,
   handleCancel,
   handleSubmit,
-}) => {
-  const {
-    completeInstructions,
-    dose,
-    dosingUnit,
-    frequency,
-    route,
-    duration,
-    durationUnit,
-    dispensingUnit,
-    dispensingQuantity,
-    reason,
-    note,
-  } = fields;
-  return (
-    <div>
-      <h2>Unsaved Draft Data</h2>
-      <table>
-        <thead>
-          <tr>
-            <td><h3>Status</h3></td>
-            <td><h3>Instructions</h3></td>
-            <td><h3>Edit</h3></td>
-          </tr>
-        </thead>
-        <tbody>
-          <tr>
-            <td>{status}</td>
+}) => (
+  <div>
+    <h2>Unsaved Draft Orders ({orders.length})</h2>
+    <table>
+      <thead>
+        <tr>
+          <td><h3>Status</h3></td>
+          <td><h3>Instructions</h3></td>
+          <td><h3>Actions</h3></td>
+        </tr>
+      </thead>
+      <tbody>
+        {orders.map(order => (
+          <tr key={order.orderNumber}>
+            <td>{order.action}</td>
             <td>
-              {drugName}:
-              {dose && `${dose}` }
-              {dosingUnit && `${dosingUnit},` }
-              {frequency && `${frequency},` }
-              {route && `${route},` }
-              {duration && `for ${duration} ${durationUnit} total,` }
-              {reason && `(Reason: ${reason}),` }
-              {note && `(Other notes: ${note}),`}
-              {dispensingUnit && `(Dispense: ${dispensingUnit} ${dispensingQuantity})`}
+              {order.drugName}:
+              {order.dose && ` ${order.dose}` }
+              {order.dosingUnit && ` ${order.dosingUnit},` }
+              {order.frequency && ` ${order.frequency},` }
+              {order.route && ` ${order.route}` }
+              {order.duration && `, for ${order.duration} ${order.durationUnit} total` }
+              {order.reason && `, (Reason: ${order.reason})` }
+              {order.note && ` (${order.note})`}
+              {order.dispensingUnit && `, (Dispense: ${order.dispensingUnit} ${order.dispensingQuantity})`}
             </td>
             <td>
-              <span className="icon-edit" />
-              <span className="icon-remove" />
+              <a href="#" onClick={() => handleEditDrugOrder(order, order.drug, order.dosingType)}>
+                <i className="icon-edit" title="Edit" />
+              </a>
+              <a href="#" onClick={() => handleDiscardOneOrder(order)}>
+                <i className="icon-remove" title="Delete" />
+              </a>
             </td>
           </tr>
-        </tbody>
-      </table>
-      <br />
-      <input type="button" onClick={handleCancel} className="button cancel" value="Discard" />
-      <input type="submit" onClick={handleSubmit} className="button confirm right" value="Confirm Save" />
-    </div>
-  );
-};
+        ))}
+      </tbody>
+    </table>
+    <br />
+    <input type="button" onClick={handleCancel} className="button cancel" value="Discard" />
```

Comment:
I think this button should change from `Discard` to `Discard All` depending on the number of drafts available.

Reply:
@flacode I don't get what you mean. Discard means you cancel all the orders in the draft table. For each order in the draft table, there is a remove/delete icon.